web/services/checks.go

```diff
@@ -69,7 +75,17 @@ Checks catalog services
 
 func (c *checksService) GetChecksCatalog() (models.ChecksCatalog, error) {
 	var checksEntity entities.CheckList
-	result := c.db.Order("payload->>'name'").Find(&checksEntity)
+	isPremiumActive, _ := c.premiumDetection.IsPremiumActive()
+
+	var result *gorm.DB
+	qb := c.db.Order("payload->>'name'")
+
+	if isPremiumActive {
+		result = c.db.Find(&checksEntity)
```

Comment:
This should be `qb.Find`, right?

Reply:
In fact, we could reduce this if/else condition to:
```
result = qb.Find(&checksEntity, datatypes.JSONQuery("payload").Equals("premium", isPremiumActive))
```

Reply:
> This should be qb.Find, right?
>
good catch! thanks.

> In fact, we could reduce this if/else condition to:
> ```
> result = qb.Find(&checksEntity, datatypes.JSONQuery("payload").Equals("premium", isPremiumActive))
> ```
>
For this I see an issue :thinking: 
doing like that means that we either load checks which are premium OR checks not premium, but in reality what we need to do is either load only the community checks or load all of the checks, premium and not premium.

```
if isPremiumActive {
	result = qb.Find(&checksEntity)
} else {
	result = qb.Find(&checksEntity, datatypes.JSONQuery("payload").Equals(false, "premium"))
}
```

Makes sense?

Reply:
Yes, makes sense. I guess that there is a more sqliish way of doing it, but well, let's move forward XD